pkg/distributor/distributor.go

```diff
@@ -482,6 +476,20 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		subservices = append(subservices, d.ingestStorageWriter)
 	}
 
+	// Register each metric only if the corresponding storage is enabled.
+	// Some queries in the mixin use the presence of these metrics as indication whether Mimir is running with ingest storage or not.
+	if cfg.IngestStorageConfig.Enabled {
+		promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_ingest_storage_enabled",
+			Help: "Whether writes are being processed via ingest storage. Always 1 if ingest storage is enabled.",
+		}).Set(1)
```

Comment:
Shouldn't we always export this one, and set value to 0, if it's not enabled?

Reply:
did it in https://github.com/grafana/mimir/pull/7809/commits/1b734cf71e8fbc03d31e73e6f1a8b477d49e3d8f